File: goal/src/main/java/dev/exit/asset/goal/ExitGoalRepository.java
package dev.exit.asset.goal;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;

import dev.exit.asset.goal.ExitGoal;

public interface ExitGoalRepository extends JpaRepository<ExitGoal, UUID> {

	List<ExitGoal> findByOwnerIdOrderByCreatedAtDesc(String ownerId);

	Optional<ExitGoal> findFirstByOwnerIdAndActiveTrueOrderByCreatedAtDesc(String ownerId);
}

File: goal/src/main/kotlin/dev/exit/asset/goal/ExitGoalRepository.kt
package dev.exit.asset.goal

import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface ExitGoalRepository : JpaRepository<ExitGoal, UUID> {

    fun findByOwnerIdOrderByCreatedAtDesc(ownerId: String): List<ExitGoal>

    fun findFirstByOwnerIdAndActiveTrueOrderByCreatedAtDesc(ownerId: String): Optional<ExitGoal>
}
